src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -15,54 +15,45 @@ internal partial class LocalRewriter
     {
         public override BoundNode VisitPatternSwitchStatement(BoundPatternSwitchStatement node)
         {
-            return PatternSwitchLocalRewriter.MakeLoweredForm(this, node);
+            return PatternSwitchLocalRewriter.Rewrite(this, node);
         }
 
-        /// <summary>
-        /// Helper class for rewriting a pattern switch statement by lowering it to a decision tree and
-        /// then lowering the decision tree to a sequence of bound statements. We inherit <see cref="DecisionTreeBuilder"/>
-        /// for the machinery to build the decision tree, and then use
-        /// <see cref="PatternSwitchLocalRewriter.LowerDecisionTree(BoundExpression, DecisionTree)"/>
-        /// recursively to produce bound nodes that implement it.
-        /// </summary>
-        private class PatternSwitchLocalRewriter : DecisionTreeBuilder
+        private class PatternSwitchLocalRewriter : BasePatternSwitchLocalRewriter
         {
-            private readonly LocalRewriter _localRewriter;
-            private readonly HashSet<LocalSymbol> _declaredTempSet = new HashSet<LocalSymbol>();
-            private readonly ArrayBuilder<LocalSymbol> _declaredTemps = ArrayBuilder<LocalSymbol>.GetInstance();
-            private readonly SyntheticBoundNodeFactory _factory;
-
             /// <summary>
-            /// Map from switch section's syntax to the lowered code for the section.
+            /// Map from switch section's syntax to the lowered code for the section. The code for a section
+            /// includes the code to assign to the pattern variables and evaluate the when clause. Since a
+            /// when clause can yield a false value, it can jump back to a label in the lowered decision dag.
             /// </summary>
-            private readonly Dictionary<SyntaxNode, ArrayBuilder<BoundStatement>> _switchSections = new Dictionary<SyntaxNode, ArrayBuilder<BoundStatement>>();
-
-            private ArrayBuilder<BoundStatement> _loweredDecisionTree = ArrayBuilder<BoundStatement>.GetInstance();
+            private Dictionary<SyntaxNode, ArrayBuilder<BoundStatement>> _switchSections => base._switchArms;
```

Comment:
>_switchSections [](start = 73, length = 15)

Consider removing and changing one caller to use `_switchArms`.

Reply:
I'd rather have this extra copy so the code uses the correct terminology in the switch statement rewriter and in the switch expression rewriter.

---
In reply to: [173193700](https://github.com/dotnet/roslyn/pull/24059#discussion_r173193700) [](ancestors = 173193700)